raspberry/bzzz/sensors/pressure_sensor.py

```diff
@@ -1,210 +1,236 @@
-from .filters import MedianFilter
-from .filters import AverageFilter
+import math
+from filters import *
 import smbus
 import time
 from ctypes import c_short
 from threading import Thread, Lock
 from .data_logger import DataLogger
 import datetime
 
-def _getShort(data, index):
-    """
-    Extract a signed 16-bit value from a data array starting at the specified index.
-    :param data: A list or tuple containing byte data.
-    :param index: The starting index in the data array.
-    :return: A signed 16-bit integer value.
-    """
-    return c_short((data[index] << 8) + data[index + 1]).value
-
-
-def _getUshort(data, index):
+"""
+Calculate altitude from pressure and temperature using the barometric formula.
+
+:param pres: Measured pressure
+:param temp: Measured temperature in Celsius
+:param reference_pressure_at_sea_level: Reference pressure at sea level
+:return: Calculated altitude in meters
+"""
+def _calculate_altitude(pres, temp, reference_pressure_at_sea_level):
+    k = 1.380649e-23 #Boltzmann's Number
+    m = 4.8e-26 #mass of one molecule of air 
+    g = 9.81 #gravity
+    alt = (((temp+273.15)*k)/((m)*g)) * math.log(reference_pressure_at_sea_level/pres)
```

Comment:
Being picky but remove hardcoded 273.15
something like 
celsius_to_kelvin = 273.15

Reply:
Variable names should be more descriptive. Also "mass of one molecule of air" should be changed to "molecular mass" (that's how it's called)